File: cmms/api/src/main/java/com/grash/model/GeneralPreferences.java
package com.grash.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.grash.model.enums.BusinessType;
import com.grash.model.enums.DateFormat;
import com.grash.model.enums.Language;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@NoArgsConstructor
@EqualsAndHashCode(exclude = "companySettings")
public class GeneralPreferences {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private Language language = Language.EN;

    private DateFormat dateFormat = DateFormat.MMDDYY;

    @ManyToOne
    private Currency currency;

    private BusinessType businessType = BusinessType.GENERAL_ASSET_MANAGEMENT;

    private String timeZone;

    private boolean autoAssignWorkOrders;

    private boolean autoAssignRequests;

    private boolean disableClosedWorkOrdersNotif;

    private boolean askFeedBackOnWOClosed = true;

    private boolean laborCostInTotalCost = true;

    private boolean woUpdateForRequesters = true;

    private boolean simplifiedWorkOrder;

    private int daysBeforePrevMaintNotification = 1;

    @OneToOne
    @JsonIgnore
    private CompanySettings companySettings;

    public GeneralPreferences(CompanySettings companySettings) {
        this.companySettings = companySettings;
    }


}

File: cmms/frontend/src/models/owns/generalPreferences.ts
import Currency from './currency';
import { SupportedLanguage } from '../../i18n/i18n';

export interface GeneralPreferences {
  id: number;
  language: SupportedLanguage;
  dateFormat: 'MMDDYY' | 'DDMMYY';
  currency: Currency;
  businessType: string;
  timeZone: string;
  daysBeforePrevMaintNotification: number;

  autoAssignWorkOrders: boolean;

  autoAssignRequests: boolean;

  disableClosedWorkOrdersNotif: boolean;

  askFeedBackOnWOClosed: boolean;

  laborCostInTotalCost: boolean;

  woUpdateForRequesters: boolean;

  simplifiedWorkOrder: boolean;
}
